apps/emqx_ds_shared_sub/src/emqx_ds_shared_sub_leader.erl

```diff
@@ -243,13 +243,54 @@ handle_event(Event, Content, State, _Data) ->
     }),
     keep_state_and_data.
 
-terminate(_Reason, _State, _Data) ->
-    ok.
+terminate(_Reason, _State, #{group_id := ShareTopicFilter, leader_claim := LeaderClaim}) ->
+    %% NOTE
+    %% Call to `disown_leadership/1` should be non-blocking, hence registry supervisor
+    %% should not be overwhelmed with stopping leader processes for too long.
+    Group = group_name(ShareTopicFilter),
+    emqx_ds_shared_sub_leader_store:disown_leadership(Group, LeaderClaim).
```

Comment:
Q: should check result for errors and log?

Reply:
It would almost always be `ok`, because this call boils down to an async batch write. But I guess adding explicit match won't hurt.